rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/connection/DisconnectionRouter.java

```diff
@@ -107,7 +107,7 @@ public void onGattConnectionStateException(BleGattException disconnectedGattExce
         onExceptionOccurred(disconnectedGattException);
     }
 
-    private void onExceptionOccurred(BleException exception) {
+    private synchronized void onExceptionOccurred(BleException exception) {
         if (exceptionOccurred == null) {
```

Comment:
Maybe we could use AtomicReference instead?

Reply:
Not sure if this would be sufficient for this use-case as the race may if a new observer is subscribing to `asValueOnlyObservable` after the check for `exceptionOccured` is made